.gitlab/source_test/linux.yml

```diff
@@ -35,7 +35,8 @@
     - popd
     - FAST_TESTS_FLAG=""
     - if [[ "$FAST_TESTS" == "true" ]]; then FAST_TESTS_FLAG="--only-impacted-packages"; fi
-    - inv -e test $FLAVORS --race --profile --rerun-fails=2 --python-runtimes "$PYTHON_RUNTIMES" --coverage --cpus $KUBERNETES_CPU_REQUEST $EXTRA_OPTS --save-result-json $TEST_OUTPUT_FILE --junit-tar "junit-${CI_JOB_NAME}.tgz" --build-stdlib $FAST_TESTS_FLAG
+    - inv -e sds.build-library
```

Comment:
❓ question: Should we also have a similar change for macOS somewhere [here](https://github.com/DataDog/datadog-agent-macos-build/blob/2fd82bbde8734b53d6d07deb13715fb8970f0a13/scripts/test_script.sh)?

Reply:
Very good point, yes !

Reply:
macOS isn't officially supported for now: I prepared the support for most of it but I'd prefer to merge the full macOS support once we actively do. For now, it builds and compiles against a mocked implementation and we don't need the library in the CI (and it doesn't run the unit tests on macOS).

Reply:
How much overhead is this adding to the job? Could we do this once per pipeline vs once per job?

Reply:
It's a 20s step (from pipeline run numbers, on ~6 minutes job) and running besides other jobs tests, and it is depending on the architecture.

I can maybe look into having separate jobs (one per arch) doing it once per pipeline, pushing the result as an artifact in the pipeline and retrieving it in the other jobs (if Gitlab support this?), but the compilation will still have to be done before all jobs can use the said artifact, i.e., the 20s step will have to be waited for before running the tests anyway. This is probably slightly better for a github-rate-limited perspective but it won't be faster.

Reply:
Ideally have it compiled and regularly integrated would be the best, this is a source of potential flakes in pipelines

Reply:
From offline conversation and thanks to @KSerrania, there are no limits to `git` requests to github, so no risk of rate limits. We also agreed on pinning the sds version, so all good for me!

Reply:
Discussed offline:
  * OK with current small additional compute time
  * Doing better will though be tracked in a card.